libs/coin-modules/coin-sui/src/api/index.ts

```diff
@@ -39,3 +41,40 @@ async function estimate(transactionIntent: TransactionIntent<SuiAsset>): Promise
   const fees = await estimateFees(transactionIntent);
   return { value: fees };
 }
+
+async function list(address: string, pagination: Pagination): Promise<[Operation<SuiAsset>[], string]> {
+  return operationsFromHeight(address, pagination.minHeight)
+}
+
+type PaginationState = {
+  readonly heightLimit: number;
+  continueIterations: boolean;
+  apiNextCursor?: string;
+  accumulator: Operation<SuiAsset>[];
+};
+
+async function operationsFromHeight(
+  address: string,
+  minHeight: number,
+): Promise<[Operation<SuiAsset>[], string]> {
+  const state: PaginationState = {
+    heightLimit: minHeight,
+    continueIterations: true,
+    accumulator: [],
+  };
+
+  while (state.continueIterations) {
+    const option = state.apiNextCursor
+      ? { minHeight: minHeight, cursor: state.apiNextCursor }
+      : { minHeight: minHeight };
+    const [operations, nextCursor] = await listOperations(address, option);
+    const opsFromHeight = operations.filter(op => op.tx.block.height >= minHeight);
+    state.accumulator.push(...opsFromHeight);
+    state.apiNextCursor = nextCursor;
+    // The API makes 2 calls "IN" and "OUT" so even if we started filtering a few elements, we may still have more txs to fetch on one of the two calls.
+    // We can only stop when we've filtered all txs.
+    state.continueIterations = nextCursor !== "" && opsFromHeight.length > 0;
```

Comment:
I think this void the interest of having this filter... What about we propagate the filter to `listOperations` so it can stop as early as possible on the sub requests ?

Reply:
it would be best to change the whole pagination algo in the logic layer, yes..

Reply:
(discussed internally: we go through with this PR, and then work on it when general integration is working)